### ISSUE TITLE
how can i decrease bandwidth? is there a cure ? is it possible to cure bandwidth to not cost money?

### ISSUE THREAD

**Issue by VSabertail** [AJS-development/Ogar-unlimited#425]
please tell me if theres a cure or how i can decrease bandwidth

**Comment by MichaelLeeHobbs** [comment]
Why do you think bandwidth is an issue?

**Comment by VSabertail** [comment]
cause it cost so much

**Comment by MichaelLeeHobbs** [comment]
Explain and not just a few words. Show me the BANDWIDTH usage/cost and not the SERVER cost. I mean AWS charges $0.16 per gig! If you did 10gb per moth that'd be $1.60. I would be very surprised if Ogar did more than a gig or two a month on a busy server.

**Comment by VSabertail** [comment]
and it does about 10gb per hour for me

**Comment by MichaelLeeHobbs** [comment]
@VSabertail Really? I don't have wireshark loaded at home right now, but I'll get it installed latter so I can do some monitoring and see what is going on. I haven't really taken a deep look at the network code as it is a mess and hard to follow. 

I'm actually thinking that our GameSever optimization might have caused it to send out packets too frequently. IF this is the case then lowering the fps in advConfig will reduce the bandwidth usage but will also reduce the number of bots/players you can have.

**Comment by VSabertail** [comment]
ok

**Comment by MichaelLeeHobbs** [comment]
@VSabertail Is there some sort of bandwidth report? I wonder if they are counting local bandwidth against you. This would be traffic going to localhost:5984 or 127.0.0.1:5984. I could easily see this eating up large chunks of data but they should not count that against you.

**Comment by VSabertail** [comment]
oh

**Comment by VSabertail** [comment]
so can you fix it?

**Comment by Andrews54757** [comment]
@MichaelLeeHobbs  I took out the db because of that in master

**Comment by Andrews54757** [comment]
so its not the problem

**Comment by MichaelLeeHobbs** [comment]
@VSabertail It all depends on what the actually problem is. You have given us very little to work on. What is the actually bandwith usage, how many players are we talking, and over what period of time? Who is your provider? Are you sure this is bandwidth usage and not CPU usage? Doing some quick calculations I would consider 30megs and hour per player reasonable. 

If the problem comes down to us sending out too many packets due to the optimizations we have made then yes it is fixable. However, it may not be an easy fix. If this is the case then just dont run more than say 10 bots and drop the FPS down to 5-10 a second.

**Comment by VSabertail** [comment]
ok

**Comment by PrinceDark9243** [comment]
Guys If you know there is a server host for Ogar Un or Other more than 4gb plz tell me if you know

**Comment by markvillar** [comment]
@MichaelLeeHobbs  > AWS charges $0.16 per gig! If you did 10gb per moth that'd be $1.60

AWS must be a good host?.. I wonder if there are any other hosts that charges less but easy as DigitalOcean to manage. DigitalOcean billed me **$0.86** for hosting Ogar-Un for **115 hours** (approximately 5 days, low usage) with their cheapest **512MB** package, and I know DigitalOcean charges at /hour rate, but that was just too expensive and impractical for these types of games in the long run.

**Comment by MichaelLeeHobbs** [comment]
@markvillar the $0.16 does not count cpu. However AWS has a free tier for up to one year that could easily support a small OUL.

**Comment by MichaelLeeHobbs** [comment]
@VSabertail You have not posted any bandwidth reports or anything else we have requested.
